### PR TITLE
Add stashvar model overwrite functionality, and missing ESM1.5 variable names

### LINKED ISSUE #157: Missing variable names
ESM1.5 configs include additional STASHmaster entries in the [`prefix.PRESM_A` file](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/release-preindustrial%2Bconcentrations/atmosphere/prefix.PRESM_A). Not all of these variables are currently in the [`stashvar_cmip6.py`](https://github.com/ACCESS-NRI/um2nc-standalone/blob/main/umpost/stashvar_cmip6.py) file used by um2nc. 

As a result, several variables in ESM1.5 output are given the long name `UNKNOWN VARIABLE`. This particularly affects many of the CASA CNP variables, e.g our converted ESM1.5 output contains:
```
	fld_s03i835:long_name = "UNKNOWN VARIABLE" ;
	fld_s03i851:long_name = "UNKNOWN VARIABLE" ;
	fld_s03i852:long_name = "UNKNOWN VARIABLE" ;
	fld_s03i853:long_name = "UNKNOWN VARIABLE" ;
       ...
```
which all correspond to CASA CNP variables.

Where a long name is available in the `prefix.PRESM_A` file, and doesn't clash with CM2, we should add it in to `stashvar_cmip6.py`.

### PR DESCRIPTION
**Pull request by blimlim**
Closes #157 and #25

This PR allows for model specific `stashvar` dictionaries to be added in `stashvar_cmip6.py`. This lets us to provide correct variable names, units, etc, in cases where CM2 and ESM1.5 disagree. E.g. `3173` refers to `QCF INCR: bdy layer negative` in CM2, and `3173 is soil (heterotrophic) respiration on tiles` in ESM1.5.

The custom dictionaries are accessed via the added `--model` command line argument, with options `ESM1PX` (for ESM1.5 and 1.6) and `CM2`. If a variable is not found in the model specific dictionary, the default `atm_stashvar` data will be used. Hence the custom dictionaries only need to contain overwrites of the defaults, and don't need to duplicate the whole `atm_stashvar` dictionary.

To implement this, I've repurposed the existing `model` argument in the `StashVar` class. This appears to have been unused, and so the changes should be safe.

The new `ESM1PX` stashvar dictionary consists of all variables in ESM1.5's [prefix.PRESM_A](https://github.com/ACCESS-NRI/access-esm1.5-configs/blob/release-historical%2Bconcentrations/atmosphere/prefix.PRESM_A) that weren't already present in the default `atm_stashvar`. Any variables with space code 3 (which can't be saved) were ignored.

I also added the `3173` field to fix the clash described in #25.

Note that this PR does not fix the incorrect `standard_name` described in [this comment](https://github.com/ACCESS-NRI/um2nc-standalone/issues/25#issuecomment-2234903318) for `33001`: `ATM TRACER  1               AFTER TS`. I believe is a tracer of CO2 from land sources, and it's not clear that a `standard_name` is available for this.

### CODE REVIEW

**Comment by blimlim** [comment]
I've just converted this to a draft – I was incorrect in assuming we can ignore the section 0 variables with space code 3. These can appear in restart files, and so we want to name them correctly too!

**Comment by atteggiani** [comment]
> I've just converted this to a draft – I was incorrect in assuming we can ignore the section 0 variables with space code 3. These can appear in restart files, and so we want to name them correctly too!

All good, just let me know when it's ready for review again please.

**Comment by blimlim** [comment]
Thanks @atteggiani, the fixes are in and this one should be ready for review!

**Comment by blimlim** [comment]
@atteggiani I've just resolved the conflicts with `main` which has let the CI run. It looks like `StrEnum`s were added in `Python3.11`, and so the `3.10` tests fail.

Would there be any major downsides of swapping `class STASHmaster(strEnum)` to a generic Enum `class STASHmaster(Enum)`, which would work with 3.10?

**Comment by atteggiani** [comment]
> @atteggiani I've just resolved the conflicts with `main` which has let the CI run. It looks like `StrEnum`s were added in `Python3.11`, and so the `3.10` tests fail.
> 
> Would there be any major downsides of swapping `class STASHmaster(strEnum)` to a generic Enum `class STASHmaster(Enum)`, which would work with 3.10?

Oh, I missed that `StrEnum` was added in `3.11`.
I think a general `Enum` can be turned to a `StrEnum` simply by inheriting from both `str` and `Enum` (there are a few more features in the actual implementation, but for our use case I think it doesn't matter).

So, replacing:

https://github.com/ACCESS-NRI/um2nc-standalone/blob/0344a4dc297c9577a98e4e5d15d833894e9ade44/src/um2nc/stashmasters.py#L4942

with:
```py
class STASHmaster(str, Enum):
```
Should do the job.

**Review by atteggiani — APPROVED** [review]
LGTM! 

Thank you @blimlim for the additions and rework! :)

File: umpost/um2netcdf.py [hunk 1]
```diff
@@ -1122,6 +1157,18 @@ def parse_args():
...
 
+    parser.add_argument(
+        "--model",
+        dest="model",
+        type=STASHmaster,
+        action=EnumAction,
+        help=(
+            "Link STASH codes to variable names and metadata by using a preset STASHmaster associated with a specific model. "
+            f"Options: {[v.value for v in STASHmaster]}. "
+            "If omitted, the 'cmip6' STASHmaster will be used."
```
[the remark this answers is not in the document]
Comment (blimlim) [reply umpost/um2netcdf.py]: Ah the comment is showing now! Added in 8a0fde1

**Pull request merged**